### DISCUSSION TITLE
Error Message on Website when running Performance scans on Google Lighthouse

### DISCUSSION THREAD

**Discussion by daphneyyt** [GoogleChrome/lighthouse#13744] [General]
Dear Lighthouse team

We are running performance scans our client for this website: [https://wponline.mom.gov.sg/WPOLOMController](url)

(To access the page, please visit [this URL](https://wponline.mom.gov.sg/WPOLLoginController?action=WPOLLoginAction&actionType=WPOLNonLogin) and select the "I Agree" button)

![MOM WPOL - 4](https://user-images.githubusercontent.com/75086091/158317364-18f5403a-2e6c-41ed-b956-7f20c2f4e722.png)

 **Issue:**
When we run Performance scans on Google Lighthouse (for the URL: "https://wponline.mom.gov.sg/WPOLOMController"), we are redirected to this error page with the following message:  

![MOM WPOL - 1](https://user-images.githubusercontent.com/75086091/158316978-19abdb9c-32ab-4e4a-bb4f-d0fa104abca1.png)
![MOM WPOL - 2](https://user-images.githubusercontent.com/75086091/158316984-28b21ab9-fa76-45b8-95cd-d9add255a423.png)

The results captured by Lighthouse is that for the error page and not the website itself.
![MOM WPOL - 3](https://user-images.githubusercontent.com/75086091/158316997-2bead393-7f33-4515-9f93-495b0d8c53b9.png)

**Questions:**
Appreciate if the Lighthouse team could:
(i) shed some insights into what is happening, and 
(ii) what the web developers should do to rectify the issue?

Thank you

**Comment by adamraine** [comment]
> shed some insights into what is happening

The destination page (https://github.com/GoogleChrome/lighthouse/issues/url) is only showing the desired content if the user arrived after clicking the "I agree" button. Lighthouse in DevTools will always navigate to the URL directly, so the Lighthouse navigation won't have the same authorization as clicking the "I agree" button.

> what the web developers should do to rectify the issue?

There are a couple options:
- Create a testing URL that displays the desired content without requiring authorization, or has the authorization provided in the destination URL.
- Use the new Lighthouse node API to navigate via user interaction. I was able to get a Lighthouse report on the destination page with this Node script:
	- Disclaimer: This is currently a preview feature, the API is subject to change.

```js
const puppeteer = require('puppeteer');
const fs = require('fs');
const open = require('open');
const lighthouse = require('lighthouse/lighthouse-core/fraggle-rock/api.js');

const config = {
  extends: 'lighthouse:default',
  settings: {
    output: 'html',
  },
};

(async () => {
  const browser = await puppeteer.launch({
    executablePath: process.env.CHROME_PATH,
    ignoreDefaultArgs: ['--enable-automation'],
    headless: false,
  })
  const page = await browser.newPage();

  await page.goto('https://wponline.mom.gov.sg/WPOLLoginController?action=WPOLLoginAction&actionType=WPOLNonLogin');
  const {report} = await lighthouse.navigation(async () => {
	// CSS selector of the "I agree" button
    await page.click('body > form > table:nth-child(14) > tbody > tr > td > input[type=submit]:nth-child(1)');
  }, {page, config});

  fs.writeFileSync('report.html', report);
  open('report.html');

  await page.close();
  await browser.close();
})();
```
